common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java

```diff
@@ -56,7 +53,7 @@
   private static final long                 MIN_RETRY_INTERVAL    = 1000;
   public static final String                RECONNECT_THREAD_NAME = "ConnectionEstablisher";
 
-  private final LinkedHashSet<ConnectionInfo>   connAddressProvider;
+  private volatile Iterable<InetSocketAddress> serverAddresses;
```

Comment:
why is this volatile?  it is set in a synchronized block.

Reply:
but not all reads under the synchronized block.

Reply:
toString reads don't matter.  I don't volatile is warranted here.

Reply:
what about reconnect(...)?